nwbwidgets/misc.py

```diff
@@ -286,7 +287,7 @@ def __init__(
         if unit_controller is None:
             nunits = len(units["spike_times"].data)
             self.unit_controller = widgets.Dropdown(
-                options=[x for x in range(nunits)],
+                options=[(str(units.id.data[x]), x) for x in range(nunits)],
```

Comment:
This looks great. The only problem is that his is an unnecessarily high number of H5 Dataset read, which will be slow if there are many units. It would be better to create an ids `np.array` first:
```suggestion
                ids = units.id.data[:]
                options=[(str(ids[x]), x) for x in range(nunits)],
```